apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplTest.cs

```diff
@@ -69,5 +76,402 @@ public void GetEffectiveEncryptionKey_Invalid(EncryptionKey clientkey, Encryptio
+
+        [Fact]
+        public void ListObject_RetryOnce() =>
+            RetryOnceHelper(service => service.Objects.List("bucket"), client => client.ListObjects("bucket").Count());
+
+        [Fact]
+        public void ListObject_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Objects.List("bucket"), client => client.ListObjects("bucket").Count());
+
+        [Fact]
+        public void DeleteObject_NoRetry() =>
+            NoRetryHelper(service => service.Objects.Delete("bucket", "objectName"), client => client.DeleteObject("bucket", "objectName"));
+
+        [Fact]
+        public void DeleteObject_RetryOnce() =>
+            RetryOnce(service => new ObjectsResource.DeleteRequest(service, "bucket", "objectName") { Generation = 70 }, 
+                client => client.DeleteObject("bucket", "objectName", new DeleteObjectOptions { Generation = 70 }), "check");
+
+        [Fact]
+        public void DeleteObject_RetryThenFail() =>
+            RetryThenFail(service => new ObjectsResource.DeleteRequest(service, "bucket", "objectName") { Generation = 70 }, 
+                client => client.DeleteObject("bucket", "objectName", new DeleteObjectOptions { Generation = 70 }), "check");
+
+        [Fact]
+        public void UpdateObject_WithMetageneration_RetryOnce() =>
+           RetryOnceHelper(
+               service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
+               client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }, new UpdateObjectOptions { IfMetagenerationMatch = 70 }));
+
+        [Fact]
+        public void UpdateObject_WithoutMetageneration_NoRetry() =>
+            NoRetryHelper(
+                service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object"),
+                client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }, new UpdateObjectOptions()));
+
+        [Fact]
+        public void UpdateObject_WithoutOptions_NoRetry() =>
+            NoRetryHelper(
+                service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object"),
+                client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }));
+
+        [Fact]
+        public void UpdateObject_RetryThenFail() =>
+            RetryThenFailHelper(service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
+                client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }, new UpdateObjectOptions { IfMetagenerationMatch = 70 }));
+
+        [Fact]
+        public void PatchObject_WithMetageneration_RetryOnce() =>
+            RetryOnceHelper(
```

Comment:
This format looks very readable to me - so let's apply it to all the tests above as well, to be consistent. (At least the ones where it doesn't all just fit on one line easily.

Another option would be to keep the method name on the same line as the test:

```csharp
[Fact]
public void PatchObject_WithMetageneration_RetryOnce() => RetryOnceHelper(
    service => new ObjectsResource.PatchRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
    client => client.PatchObject(new Object { Bucket = "bucket", Name = "object" }, new PatchObjectOptions { IfMetagenerationMatch = 70 }));
```